src/diagnostic/detail/h5writer.hpp

```diff
@@ -133,19 +133,36 @@ class Writer
     template<typename Dict>
     static void writeAttributeDict(HighFiveFile& h5, Dict dict, std::string path)
     {
-        dict.visit([&](std::string const& key, const auto& val) {
-            h5.write_attributes_per_mpi(path, key, val);
+        dict.visit([&](std::string const& key, auto const& val) {
+            auto constexpr static unsupported
+                = std::is_same_v<std::decay_t<decltype(val)>, std::vector<std::string>>;
+
+            // the dict might have types that are not supported, but not actually contain
+            //  any of the types at runtime
+            if constexpr (!unsupported)
+                h5.write_attributes_per_mpi(path, key, val);
+
+            // runtime detection of unsupported types
+            if (unsupported)
+                throw std::runtime_error("Unsupported operation: Cannot write attribute: path("
+                                         + path + "), key(" + key + ")");
         });
     }
 
+    template<typename Dict>
+    static void writeGlobalAttributeDict(HighFiveFile& h5, Dict dict, std::string path)
+    {
```

Comment:
this one is for when all values are identical across mpi nodes, there is probably a better name

Reply:
> rename the MPI one?

they both use MPI :\

Reply:
the first calls `per_mpi` the other not, I don't understand

Reply:
they both do collective calls, just the `global` (or not `per_mpi`) version expects all values to be identical